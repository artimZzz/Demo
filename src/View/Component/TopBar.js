import React, { Component, PureComponent } from 'react';
import { ViewPropTypes, StyleSheet, Animated, View } from 'react-native';
import TouchableDebounce from './TouchableDebounce';
import Ionicons from 'react-native-vector-icons/Ionicons';
import PropTypes from 'prop-types';
import { ifAndroid, ifIPhoneX } from '../../Controller/Device/DeviceCheck';

export default class TopBar extends PureComponent {

    /* render */
    render() {
        let leftButton = this.props.leftButton ? this.props.leftButton : <View />;
        if (this.props.backCallback) {
            leftButton = (
                <TouchableDebounce style={styles.back_touch} onPress={this.props.backCallback}>
                    <Ionicons name='ios-arrow-back' style={styles.back_icon} size={30} />
                </TouchableDebounce>
            );
        }
        let rightButton = this.props.rightButton ? this.props.rightButton : <View />;
        let children = this.props.children ? this.props.children : <View />;
        return (
            <Animated.View style={[styles.app_top_bar, this.props.style]}>
                <View style={styles.left_button}>{leftButton}</View>
                {children}
                <View style={styles.right_button}>{rightButton}</View>
            </Animated.View>
        );
    }
}

/* style */
const styles = StyleSheet.create(
    {
        app_top_bar: {
            alignItems: 'center',
            justifyContent: 'center',
            backgroundColor: '#eee',
            ...ifAndroid(
                { paddingTop: 25 },
                ifIPhoneX({ paddingTop: 40 }, { paddingTop: 25 })
            ),
            paddingBottom: 5,
            width: '100%'
        },
        back_touch: {
            width: '100%',
            paddingRight: 30
        },
        back_icon: {
            color: 'white'
        },
        left_button: {
            position: 'absolute',
            ...ifAndroid(
                { paddingTop: 25 },
                ifIPhoneX({ paddingTop: 40 }, { paddingTop: 25 })
            ),
            left: 10
        },
        right_button: {
            position: 'absolute',
            ...ifAndroid(
                { paddingTop: 25 },
                ifIPhoneX({ paddingTop: 40 }, { paddingTop: 25 })
            ),
            right: 10
        }
    }
);

TopBar.propTypes = {
    backCallback: PropTypes.func,
    style: ViewPropTypes.style
};
TopBar.defaultProps = {
    style: {}
};