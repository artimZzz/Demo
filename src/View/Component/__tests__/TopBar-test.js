// __tests__/TopBar-test.js
import React from 'react';
import TopBar from '../TopBar';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<TopBar />).toJSON();
    expect(tree).toMatchSnapshot();
});