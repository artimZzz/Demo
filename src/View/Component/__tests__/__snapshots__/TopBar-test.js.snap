// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "justifyContent": "center",
      "paddingBottom": 5,
      "paddingTop": 25,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "left": 10,
        "paddingTop": 25,
        "position": "absolute",
      }
    }
  >
    <View />
  </View>
  <View />
  <View
    style={
      Object {
        "paddingTop": 25,
        "position": "absolute",
        "right": 10,
      }
    }
  >
    <View />
  </View>
</View>
`;
